### PR TITLE
Fix performance degradation during parts selection for merge

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog

Fix performance degradation in `selectPartsToMerge` introduced in #14494.

### CODE REVIEW

**Comment by alesapin** [comment]
It was not released, so nothing affected.

**Comment by alesapin** [comment]
Performance report https://clickhouse-test-reports.s3.yandex.net/14834/6f8da7dfdf3c0f8a9af9c973dcd5b5c87a0479ac/performance_comparison/report.html (for some reason no status)

File: src/Compression/CompressionCodecDoubleDelta.cpp [hunk 1]
```diff
@@ -327,18 +327,14 @@ UInt8 getDataBytesSize(DataTypePtr column_type)
 CompressionCodecDoubleDelta::CompressionCodecDoubleDelta(UInt8 data_bytes_size_)
     : data_bytes_size(data_bytes_size_)
 {
+    setCodecDescription("DoubleDelta", {});
```
Comment (Akazz) [inline src/Compression/CompressionCodecDoubleDelta.cpp:330]: Why leave empty braces here? (as well as for other codecs below)

File: src/Compression/ICompressionCodec.cpp [hunk 2]
```diff
@@ -15,24 +16,56 @@ namespace ErrorCodes
...
+            codec_desc->children.push_back(argument);
         result->arguments = codec_desc;
     }
     else
     {
+        ASTPtr codec_desc;
+        if (arguments.empty())
+            codec_desc = std::make_shared<ASTIdentifier>(codec_name);
+        else
+            codec_desc = makeASTFunction(codec_name, arguments);
```
[the remark this answers is not in the document]
Comment (alesapin) [reply src/Compression/ICompressionCodec.cpp:41]: No, it's different. Lines 31-32 is a special case for codec multiple which just `ASTExpressionList` of different codecs. It's not equal to ASTFunction with an empty name. And yes, we have codecs like `CODEC(LZ4HC(5))`.

**Review by Akazz — APPROVED** [review]

**Pull request merged**